### ISSUE TITLE
[C#] Compile error - Getter tries to return null for newly introduced Enum property

### ISSUE THREAD

**Issue by EddPorter** [real-logic/simple-binary-encoding#661]
When a template specification introduces a new property with an Enum return type, the generated code checks the ActingVersion and outputs "return null" if it is too low. This doesn't compile.

```
public AttribValue AttribValue
{
  get
  {
    if (_actingVersion < 3) return null; // compile error

    return (AttribValue)_buffer.Uint32GetLittleEndian(_offset + 0);
  }
... setter
}
```

From schema file:
```
<set name="AttribValue" encodingType="uInt32" sinceVersion="3">
            <choice name="FirstChoice" description="The first choice">0</choice>
...
</set>
```
Should the return type be `AttribValue?` as a fix?

**Comment by billsegall** [comment]
I'm thinking it should return (AttribValue)0. This isn't ideal but I've failed to come up with anything better.

As it shouldn't be reference in code using older versions I think it will be ok in practice.

**Comment by mjpt777** [comment]
Java supports the concept of `SBE_UNKNOWN` which can be enabled which can be enabled with the system property `sbe.decode.unknown.enum.values` for SbeTool. If not supported then maybe throw an exceptions.

**Comment by EddPorter** [comment]
I think this is resolved by #663
